src/main/scala/org/apache/mesos/chronos/scheduler/jobs/JobScheduler.scala

```diff
@@ -609,10 +610,12 @@ class JobScheduler @Inject()(val scheduleHorizon: Period,
 
   //Begin Leader interface, which is required for CandidateImpl.
   def onDefeated() {
-    mesosDriver.close()
-
-    log.info("Defeated. Not the current leader.")
+    import scala.concurrent.ExecutionContext.Implicits.global
```

Comment:
Like in https://github.com/Rob-Johnson/chronos/blob/die-on-loss-of-zk-connection/src/main/scala/org/apache/mesos/chronos/scheduler/jobs/JobScheduler.scala#L48, I recommend creating a `val exitExecutor = Executors.newSingleThreadExecutor()`. You can adapt this Java `Executor` to a Scala `ExecutionContext` by `ExecutionContext.fromExecutor(exitExecutor)` for you `RichRuntime`.